server/plugins/schema-plugin-ob-mysql/src/main/java/com/oceanbase/odc/plugin/schema/obmysql/parser/OBMySQLGetDBTableByParser.java

```diff
@@ -241,36 +236,140 @@ public DBTablePartition getPartition() {
+         *        );
+         *
+         * In this case, the subpartitionsNum is not fixed.
+         * </pre>
+         */
+        subPartitionOption
+                .setPartitionsNum(
+                        parsedSubPartitionOption.getTemplates() != null ? parsedSubPartitionOption.getTemplates().size()
+                                : null);
+        for (PartitionElement partitionElement : partitionStmt.getPartitionElements()) {
+            if (partitionElement.getSubPartitionElements() != null) {
+                // obtain DBTablePartitionDefinitions for non-templated subpartitions
+                for (int i = 0; i < partitionElement.getSubPartitionElements().size(); i++) {
+                    DBTablePartitionDefinition subPartitionDefinition = new DBTablePartitionDefinition();
+                    SubPartitionElement subPartitionElement = partitionElement.getSubPartitionElements().get(i);
+                    fillSubPartitionValue(subDBTablePartitionType, subPartitionElement, subPartitionDefinition);
+                    subPartitionDefinition.setName(
+                            removeIdentifiers(subPartitionElement.getRelation()));
+                    subPartitionDefinition.setOrdinalPosition(i);
+                    subPartitionDefinition.setType(subDBTablePartitionType);
+                    subPartitionDefinitions.add(subPartitionDefinition);
+                }
+            } else {
+                // obtain DBTablePartitionDefinitions for templated subpartitions
+                String parentPartitionName = removeIdentifiers(partitionElement.getRelation());
+                List<SubPartitionElement> templates = partitionStmt.getSubPartitionOption().getTemplates();
+                for (int i = 0; i < templates.size(); i++) {
+                    DBTablePartitionDefinition subPartitionDefinition = new DBTablePartitionDefinition();
+                    SubPartitionElement subPartitionElement = templates.get(i);
+                    fillSubPartitionValue(subDBTablePartitionType, subPartitionElement, subPartitionDefinition);
+                    // for a templated subpartition table, the naming rule for the subpartition is
+                    // '($part_name)s($subpart_name)'.
+                    subPartitionDefinition.setName(
+                            parentPartitionName + 's' + removeIdentifiers(subPartitionElement.getRelation()));
+                    subPartitionDefinition.setOrdinalPosition(i);
+                    subPartitionDefinition.setType(subDBTablePartitionType);
+                    subPartitionDefinitions.add(subPartitionDefinition);
+                }
+            }
+        }
+    }
+
+    private void fillSubPartitionValue(DBTablePartitionType subDBTablePartitionType,
+            SubPartitionElement subPartitionElement,
+            DBTablePartitionDefinition subPartitionDefinition) {
+        if (subDBTablePartitionType == DBTablePartitionType.LIST) {
```

Comment:
don't use DBTablePartitionType, use `instance of` instead, it is more reliable:

```Java
if (subPartitionElement instanceof SubListPartitionElement) {
    ....
}
```

Reply:
The object of the SubListPartitionElement still has to judge the processing logic against the subDBTablePartitionType. Because SubListPartitionElement contains both LIST and LIST_COLUMNS

Reply:
ok